verification/src/solidity/verifier.rs

```diff
@@ -97,3 +327,329 @@ impl Verifier {
+        DEFAULT_ENCODED_METADATA_HASH,
+        DEFAULT_CONSTRUCTOR_ARGS
+    );
+    const DEFAULT_DEPLOYED_BYTECODE: &'static str = concatcp!(
+        DEFAULT_DEPLOYED_BYTECODE_WITHOUT_METADATA_HASH,
+        DEFAULT_ENCODED_METADATA_HASH
+    );
+
+    #[test]
+    fn test_initialization_with_valid_data() {
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            DEFAULT_CREATION_TX_INPUT,
+            DEFAULT_DEPLOYED_BYTECODE,
+        );
+        assert!(
+            verifier.is_ok(),
+            "Initialization without \"0x\" prefix failed"
+        );
+
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            &concatcp!("0x", DEFAULT_CREATION_TX_INPUT),
+            &concatcp!("0x", DEFAULT_DEPLOYED_BYTECODE),
+        );
+        assert!(verifier.is_ok(), "Initialization with \"0x\" prefix failed");
+    }
+
+    #[test]
+    fn test_initialization_with_empty_creation_tx_input_should_fail() {
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            "",
+            DEFAULT_DEPLOYED_BYTECODE,
+        );
+        assert!(verifier.is_err(), "Verifier initialization should fail");
+        assert_eq!(
+            verifier.unwrap_err(),
+            InitializationError::MetadataHashMismatch(Mismatch::expected(
+                DisplayBytes::from_str(DEFAULT_ENCODED_METADATA_HASH).unwrap()
+            ))
+        )
+    }
+
+    #[test]
+    fn test_initialization_with_creation_tx_input_as_invalid_hex_should_fail() {
+        let invalid_input = "0xabcdefghij";
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            invalid_input,
+            DEFAULT_DEPLOYED_BYTECODE,
+        );
+        assert!(verifier.is_err(), "Verifier initialization should fail");
+        assert_eq!(
+            verifier.unwrap_err(),
+            InitializationError::InvalidCreationTxInput
+        )
+    }
+
+    #[test]
+    fn test_initialization_with_empty_deployed_bytecode_should_fail() {
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            DEFAULT_CREATION_TX_INPUT,
+            "",
+        );
+        assert!(verifier.is_err(), "Verifier initialization should fail");
+        assert_eq!(
+            verifier.unwrap_err(),
+            InitializationError::MetadataHashParseError("length is not encoded".to_string())
+        )
+    }
+
+    #[test]
+    fn test_initialization_with_deployed_bytecode_as_invalid_hex_should_fail() {
+        let invalid_input = "0xabcdefghij";
+        let verifier = Verifier::new(
+            DEFAULT_CONTRACT_NAME.to_string(),
+            None,
+            DEFAULT_CREATION_TX_INPUT,
+            invalid_input,
+        );
+        assert!(verifier.is_err(), "Verifier initialization should fail");
```

Comment:
`expect_err`?

Reply:
I prefer using an assertion here as verification failure is the part of what we are actually testing. `expect_err` mostly indicates that we are sure that the result is error due to some preconditions and we just want to get the internal error. Here it is not the case